### PR TITLE
feat(getCurrentInstance): Aligning with vue3

### LINKED ISSUE #499: Alias `vm.$root` with `this.root` to match with Vue 3
https://github.com/vuejs/vue-apollo/pull/1040#issuecomment-685866246 by @frandiox

### PR DESCRIPTION
**Pull request by pikax**
fix #499 

BREAKING CHANGE: The internal vm can be accessed with `getCurrentInstance().proxy`

```js
const vm = getCurrentInstance()

// becomes

const vm = getCurrentInstance().proxy
```

### CODE REVIEW

**Comment by antfu** [comment]
I was just planning to file an RFC on this... Great, thanks 👍

File: src/runtimeContext.ts [hunk 1]
```diff
@@ -57,10 +57,148 @@ export function setVueConstructor(Vue: VueConstructor) {
...
+  'attrs',
+  'refs',
+  'emit',
+  'vnode',
+] as const
+
+export function getCurrentInstance() {
+  const internal = getCurrentInternalInstance()
+  if (internal) {
+    return getComponentInstanceVue3(internal)
```
Comment (antfu) [inline src/runtimeContext.ts:133]: Maybe `getCurrentVue2Instance` and `toVue3Instance`?

File: src/runtimeContext.ts [hunk 2]
```diff
@@ -57,10 +57,148 @@ export function setVueConstructor(Vue: VueConstructor) {
...
+}
+
+const specialProps = [
+  'data',
+  'props',
+  'attrs',
+  'refs',
+  'emit',
+  'vnode',
+] as const
```
Comment (antfu) [inline src/runtimeContext.ts:128]: I guess `root` and `parent` can be included as well.

**Comment by antfu** [comment]
@knightly-bot build this

**Comment by antfu** [comment]
@pikax I think I am done, could you have a look? Thanks

**Comment by pikax** [comment]
LGTM ✔️

**Comment by antfu** [comment]
Thanks! Heading for the release.

**Pull request merged**